### PR TITLE
reformat for sphinx httpdoc

### PR DESCRIPTION
**Pull request by andymckay**

### CODE REVIEW

**Comment by andymckay** [comment]
screenshot 2015-11-03 15 22 31

**Comment by andymckay** [comment]
Sticks an annoying link on the end of the URL, not sure why.

File: docs/conf.py [hunk 1]
```diff
@@ -49,9 +49,9 @@
 # built documents.
 #
 # The short X.Y version.
-version = '0.8'
+version = '3.0'
 # The full version, including alpha/beta/rc tags.
-release = '0.8'
+release = '3.0'
```
Comment (andymckay) [inline docs/conf.py]: We can probably stop calling this 0.8 of olympia right?

Comment (mstriemer) [inline docs/conf.py]: Should it just be `3` or does it need the minor version?

Comment (andymckay) [inline docs/conf.py]: Let's go wild.

File: docs/topics/api/signing.rst [hunk 2]
```diff
@@ -14,15 +14,33 @@ as multi-part formdata. This will create a pending version on the
...
+    :reqheader Content-Type: multipart/form-data
+
+    **Response:**
+
+    The response body will be the same as the :ref:`check-status` response.
+
+    :statuscode 201: new add-on and version created.
+    :statuscode 202: new version created.
+    :statuscode 400: an error occurred, check the `error` value in the JSON.
+    :statuscode 403: authentication failed, or you do not own this add-on.
```
Comment (andymckay) [inline docs/topics/api/signing.rst]: Actually auth failed is a 401?

Comment (mstriemer) [inline docs/topics/api/signing.rst]: You should be able to get both. 401 should happen if your jwt is invalid or missing while 403 will happen if you're authenticated but don't have permissions to this add-on.

Comment (mstriemer) [inline docs/topics/api/signing.rst]: So yes, auth failed should be 401.

Comment (andymckay) [inline docs/topics/api/signing.rst]: Fixed

File: docs/topics/api/signing.rst [hunk 3]
```diff
@@ -14,15 +14,34 @@ as multi-part formdata. This will create a pending version on the
...
+            -XPUT --form 'upload=@build/my-addon.xpi' -H 'Authorization: JWT <jwt-token>'
+
+    :param addon-guid: the GUID for the add-on.
+    :param version: the version of the add-on.
+    :form upload: the add-on being uploaded.
+    :reqheader Content-Type: multipart/form-data
+
+    **Response:**
+
+    The response body will be the same as the :ref:`check-status` response.
```
Comment (mstriemer) [inline docs/topics/api/signing.rst]: Do you know if these refs are global? I was originally thinking they were local to the file but I have a feeling they aren't maybe this should be ``:ref:`version-status` `` instead?

File: docs/topics/api/signing.rst [hunk 4]
```diff
@@ -14,15 +14,34 @@ as multi-part formdata. This will create a pending version on the
...
+
+    :statuscode 201: new add-on and version created.
+    :statuscode 202: new version created.
+    :statuscode 400: an error occurred, check the `error` value in the JSON.
+    :statuscode 401: authentication failed.
+    :statuscode 403: you do not own this add-on.
+    :statuscode 404: add-on not found.
```
Comment (mstriemer) [inline docs/topics/api/signing.rst]: I guess `404` won't happen anymore since we'll create it.

File: docs/topics/api/signing.rst [hunk 5]
```diff
@@ -55,29 +74,55 @@ longer. Once review is complete then the ``reviewed`` property
 will be set and you can check the results with the ``passed_review``
 property.
 
-To make the request using curl::
+To make the request using curl:
```
Comment (mstriemer) [inline docs/topics/api/signing.rst]: Looks like this line should move down a bit.

File: docs/topics/api/signing.rst [hunk 6]
```diff
@@ -55,29 +74,55 @@ longer. Once review is complete then the ``reviewed`` property
...
-        "url": "https://addons.mozilla.org/en-US/firefox/api/v3/addons/%40new-unlisted-api/versions/0.0.0/",
-        "valid": true,
-        "validation_results": {
-            ... snip ...
-        },
-        "validation_url": "https://addons.mozilla.org/en-US/developers/upload/f68abbb3b1624c098fe979a409fe3ce9",
-        "version": "0.0.0"
-    }
+
+    :>json active: add-on is active.
```
Comment (mstriemer) [inline docs/topics/api/signing.rst]: "version is active."

File: docs/topics/api/signing.rst [hunk 7]
```diff
@@ -55,29 +74,55 @@ longer. Once review is complete then the ``reviewed`` property
...
+    :>json files.download_url: URL to download the add-on file.
+    :>json files.signed: if the file is signed.
+    :>json passed_review: if the add-on has passed review.
```
Comment (mstriemer) [inline docs/topics/api/signing.rst]: These `add-on`s should be `version` too, I think.

Comment (mstriemer) [inline docs/topics/api/signing.rst]: Although I guess I see this from an `add-on` versus `Addon` viewpoint also.

File: docs/topics/api/signing.rst [hunk 8]
```diff
@@ -55,29 +74,55 @@ longer. Once review is complete then the ``reviewed`` property
...
-    {
-        "active": true,
-        "files": [
             {
-                "download_url": "https://addons.mozilla.org/firefox/downloads/file/100/unlisted_wat-0.0.0-fx+an.xpi?src=api",
-                "signed": true
+                "active": true,
+                "files": [
+                    {
+                        "download_url": "https://addons.mozilla.org/firefox/downloads/file/100/unlisted_wat-0.0.0-fx+an.xpi?src=api",
```
Comment (mstriemer) [inline docs/topics/api/signing.rst]: I probably should have sanitized the filename and guid so they weren't so silly.

File: docs/topics/api/signing.rst [hunk 9]
```diff
@@ -55,29 +74,55 @@ longer. Once review is complete then the ``reviewed`` property
...
+                        "signed": true
+                    }
+                ],
+                "passed_review": true,
+                "processed": true,
+                "reviewed": true,
+                "url": "https://addons.mozilla.org/en-US/firefox/api/v3/addons/%40new-unlisted-api/versions/0.0.0/",
```
Comment (mstriemer) [inline docs/topics/api/signing.rst]: The `version` below is `1.0` but it's `0.0.0` in the `url`.

File: docs/topics/api/signing.rst [hunk 10]
```diff
@@ -55,29 +74,55 @@ longer. Once review is complete then the ``reviewed`` property
...
+    :>json url: URL to this end point.
+    :>json valid: if the add-on passed validation.
+    :>json validation_results: the validation results (removed from the example for brevity).
+    :>json validation_url: a URL to the validation results in HTML format.
+    :>json version: the version.
+
+    :statuscode 200: request successful.
+    :statuscode 401: authentication failed.
+    :statuscode 403: you do not own this add-on.
+    :statuscode 404: add-on not found.
```
Comment (mstriemer) [inline docs/topics/api/signing.rst]: I think this would be "add-on or version not found."

**Comment by mstriemer** [comment]
That link is weird. It also stripped the trailing `/`.

Thanks for cleaning this up!

r+wc

File: docs/topics/api/signing.rst (hunk 3 above)
Comment (andymckay) [inline docs/topics/api/signing.rst]: Global.

File: docs/topics/api/signing.rst (hunk 5 above)
Comment (andymckay) [inline docs/topics/api/signing.rst]: Ah yeah, removed.

File: docs/topics/api/signing.rst (hunk 6 above)
Comment (andymckay) [inline docs/topics/api/signing.rst]: Ta

File: docs/topics/api/signing.rst (hunk 7 above)
Comment (andymckay) [inline docs/topics/api/signing.rst]: Let's go with version for precision.

File: docs/topics/api/signing.rst (hunk 9 above)
Comment (andymckay) [inline docs/topics/api/signing.rst]: Making it 1.0 throughout.

**Pull request merged**